care/utils/assetintegration/onvif.py

```diff
@@ -39,19 +50,115 @@ def handle_action(self, action):
             **action_data,
         }
 
+        if action_type == BaseAssetIntegration.BaseAssetActions.REQUEST_ACCESS.value:
+            return self.request_access(username, asset_id)
+
+        if action_type == BaseAssetIntegration.BaseAssetActions.UNLOCK_ASSET.value:
+            if self.unlock_camera(username, asset_id):
+                return {"message": "Asset Unlocked"}
+            self.raise_conflict(asset_id=asset_id)
+
+        if action_type == BaseAssetIntegration.BaseAssetActions.LOCK_ASSET.value:
+            if self.lock_camera(username, asset_id):
+                return {"message": "Asset Locked"}
+            self.raise_conflict(asset_id=asset_id)
+
         if action_type == self.OnvifActions.GET_CAMERA_STATUS.value:
+            self.lock_camera(username, asset_id)
             return self.api_get(self.get_url("status"), request_body)
 
         if action_type == self.OnvifActions.GET_PRESETS.value:
+            self.lock_camera(username, asset_id)
             return self.api_get(self.get_url("presets"), request_body)
 
         if action_type == self.OnvifActions.GOTO_PRESET.value:
-            return self.api_post(self.get_url("gotoPreset"), request_body)
+            if self.verify_access(username, asset_id):
+                self.lock_camera(username, asset_id)
+                return self.api_post(self.get_url("gotoPreset"), request_body)
+            self.raise_conflict(asset_id=asset_id)
 
         if action_type == self.OnvifActions.ABSOLUTE_MOVE.value:
-            return self.api_post(self.get_url("absoluteMove"), request_body)
+            if self.verify_access(username, asset_id):
+                self.lock_camera(username, asset_id)
+                return self.api_post(self.get_url("absoluteMove"), request_body)
+            self.raise_conflict(asset_id=asset_id)
 
         if action_type == self.OnvifActions.RELATIVE_MOVE.value:
-            return self.api_post(self.get_url("relativeMove"), request_body)
+            if self.verify_access(username, asset_id):
+                self.lock_camera(username, asset_id)
+                return self.api_post(self.get_url("relativeMove"), request_body)
+            self.raise_conflict(asset_id=asset_id)
 
         raise ValidationError({"action": "invalid action type"})
+
+    def validate_action(self, action):
+        from care.facility.models.bed import AssetBed, ConsultationBed
+
+        action_type = action["type"]
+        action_data = action.get("data", {})
+        asset_bed_id = action_data.get("id", None)
+
+        if not asset_bed_id or action_type != self.OnvifActions.RELATIVE_MOVE.value:
+            return
```

Comment:
why are we skipping the validations?

Reply:
Done, please check now

Reply:
This is used to ensure that the boundary checks are skipped if the action is not relative_move or asset_bed_id isn't given